core/test/unit/models/post_spec.js

```diff
@@ -14,6 +14,244 @@ const should = require('should'),
+                    queries[1].bindings.should.eql([
+                        false,
+                        'published',
+                        'photo',
+                        0,
+                        'public',
+                        15
+                    ]);
+                });
+            });
+
+            it('generates correct query for - filter: primary_author:leslie, with related: authors', function () {
+                const queries = [];
+                tracker.install();
+
+                tracker.on('query', (query) => {
+                    queries.push(query);
+                    query.response([]);
+                });
+
+                return models.Post.findPage({
+                    filter: 'primary_author:leslie',
+                    withRelated: ['authors']
+                }).then(() => {
+                    queries.length.should.eql(2);
+                    queries[0].sql.should.eql('select count(distinct posts.id) as aggregate from `posts` left outer join `posts_authors` on `posts_authors`.`post_id` = `posts`.`id` left outer join `users` as `authors` on `posts_authors`.`author_id` = `authors`.`id` where (`posts`.`page` = ? and `posts`.`status` = ?) and ((`authors`.`slug` = ? and `posts_authors`.`sort_order` = ? and `authors`.`visibility` = ?))');
+                    queries[0].bindings.should.eql([
+                        false,
+                        'published',
+                        'leslie',
+                        0,
+                        'public'
+                    ]);
+
+                    queries[1].sql.should.eql('select `posts`.* from `posts` left outer join `posts_authors` on `posts_authors`.`post_id` = `posts`.`id` left outer join `users` as `authors` on `posts_authors`.`author_id` = `authors`.`id` where (`posts`.`page` = ? and `posts`.`status` = ?) and ((`authors`.`slug` = ? and `posts_authors`.`sort_order` = ? and `authors`.`visibility` = ?)) group by `posts`.`id` order by CASE WHEN posts.status = \'scheduled\' THEN 1 WHEN posts.status = \'draft\' THEN 2 ELSE 3 END ASC,CASE WHEN posts.status != \'draft\' THEN posts.published_at END DESC,posts.updated_at DESC,posts.id DESC limit ?');
+                    queries[1].bindings.should.eql([
+                        false,
+                        'published',
+                        'leslie',
+                        0,
+                        'public',
+                        15
+                    ]);
+                });
+            });
+        });
+
+        describe('bad behavior', function () {
+            it('generates correct query for - filter: status:[published,draft], limit of: all', function () {
```

Comment:
@kirrg001, wasn't sure how to substitute [this test](https://github.com/TryGhost/Ghost/pull/9940/files#diff-1c8b47017aaf9d60ace1c68bbf82a5e3L1174) is this the way you had on your mind or [this](https://github.com/TryGhost/Ghost/pull/9940/files#diff-5a0b10849d1a90717049d92c728caaa5R332). 

As a side note, it's still somewhat unclean that [draft](https://github.com/TryGhost/Ghost/pull/9940/files#diff-5a0b10849d1a90717049d92c728caaa5R237) option get's to the query stage :confused:

Reply:
>  it's still somewhat unclean that draft option get's to the query stage

that's weird, why ist that?

why not unit testing https://github.com/TryGhost/Ghost/blob/master/core/server/models/post.js#L460?
the test is a public api test, which ensures you can't fetch draft posts. if we test these two model fn's, we should be fine?

Reply:
Which "these two" specifically do you have on mind? The 
 link: https://github.com/TryGhost/Ghost/blob/master/core/server/models/post.js#L460, points to `return null` in :
```
    defaultFilters: function defaultFilters(options) {
        if (options.context && options.context.internal) {
            return null;
        }

        return options.context && options.context.public ? 'page:false' : 'page:false+status:published';
    }
```
:confused: 

Will continue with this thing tomorrow think my head is not processing any more stuff today ...

Reply:
```
 enforcedFilters: function enforcedFilters(options) {
        return options.context && options.context.public ? 'status:published' : null;
    },
    defaultFilters: function defaultFilters(options) {
        if (options.context && options.context.internal) {
            return null;
        }

        return options.context && options.context.public ? 'page:false' : 'page:false+status:published';
    }
```

Reply:
@gargol These two

Reply:
added unit tests for the 2 above in fbc0eaadc98efccc0ae75f22f9d58d9ee3080456
investigating:
> it's still somewhat unclean that draft option get's to the query stage